docs/faucets/dd.md

```diff
@@ -0,0 +1,363 @@
+
+## Implementation
+
+### TestERC20
+
+Both the fee and staking assets will be instances of `TestERC20`.
+
+We'll remove the `ownerOrFreeForAll` modifier from the `mint` function, and instead add a `minters` mapping.
+
+```solidity
+contract TestERC20 is ERC20, IMintableERC20, Ownable {
+
+  mapping(address => bool) public minters;
+
+  modifier onlyMinter() {
+    require(minters[msg.sender] || msg.sender == owner(), "Not authorized to mint");
+    _;
+  }
+
+	constructor(string memory _name, string memory _symbol, address _owner)
+    ERC20(_name, _symbol)
+    Ownable(_owner)
+  {
+    minters[_owner] = true;
+  }
+
+  function addMinter(address _minter) external onlyMinter {
+    require(_minter != address(0), "Invalid address");
+    minters[_minter] = true;
+  }
+
+	function removeMinter(address _minter) external onlyMinter {
+    require(_minter != owner(), "Cannot remove owner as minter");
+    minters[_minter] = false;
+  }
+
+  function mint(address _recipient, uint256 _amount) external onlyMinter {
+    _mint(_recipient, _amount);
+  }
+}
+```
+
+### Fee Asset
+
+When the fee asset is deployed, the owner will be an EOA, held by Aztec Labs.
+
+### Fee Asset Handler
+
+For the fee asset, we will create a new contract `FeeAssetHandler`
+
+```solidity
+interface IFeeAssetHandler {
+	function mint(address _recipient) external;
+	function setMintAmount(uint256 _amount) external;
+}
+
+contract FeeAssetHandler is IFeeAssetHandler, Ownable {
+	IMintableERC20 public immutable FEE_ASSET;
+	uint256 public mintAmount;
+
+	constructor(address _owner, address _feeAsset, uint256 _mintAmount) Ownable(_owner) {
+		FEE_ASSET = IMintableERC20(_feeAsset);
+		mintAmount = _mintAmount;
+	}
+
+	function mint(address _recipient) external override {
+		FEE_ASSET.mint(_recipient, mintAmount);
+	}
+
+	function setMintAmount(uint256 _amount) external override onlyOwner {
+    mintAmount = _amount;
+	}
+}
+```
+
+We will manually deploy this contract. The owner will be the same as the current owner of the FeeAsset.
+
+The handler's deployed address will need to be made known in discord and other public channels/docs.
+
+#### Choosing the mint amount
+
+As mentioned above, we only need to prevent infinite minting (i.e. over 2^256).
+
+We can calculate the maximum mint amount based on the expected L1 block production rate, the expected L1 gas cost of a mint, and the amount of time we expect the handler to be live.
+
+Assume the cost to mint is 5K gas.
+
+Then if a block were completely full of `mint` transactions, the maximum number of mints would be 38M / 5K = 7600.
+
+This means that we could be maximally minting 7600 / 12 = 633 times per second.
+
+Suppose that we expect the handler to be live for a maximum of 5 years.
+
+Then we have a maximum of 5 \* 365 \* 24 \* 60 \* 60 \* 633 ~= 1e11 mints.
+
+Thus, the maximum mint amount is (2^256 - 1) / 1e11 ~= 1e66.
+
+But we'll err on the side of caution and use 1e42.
```

Comment:
Disagree on numbers here. I don't think you would want to ever reach the limit. You don't want to have close to max if you can get away with it, it is just pain for if the token is used in any math. I think we should target a much smaller, that is still enough to cover most everyday things. 1e42 is minting 1e24 or a TRILLION TRILLIONS. I think we can be much more sane with just a 10K or something like that. We just need to evaluate what a few things would cost.

The thing is:
- argue that we can pick a sane number
- and that it will never overflow and cause pain in any meaningful reality